JAICore/jaicore-ml/src/jaicore/ml/WekaUtil.java

```diff
@@ -790,6 +803,92 @@ public static ArrayNode splitToJsonArray(final Collection<Integer>[] splitDecisi
 		return instances;
 	}
 
+	
+	/**
+	 * Creates a StratifiedSplit for a given {@link ReproducibleInstances} Object. THe History will be updated to track the split.
+	 * 
+	 * @param data - Input data
+	 * @param rand - random used to get a seed, which can be used and saved
+	 * @param portions - ratios to split
+	 * @return a List of {@link ReproducibleInstances}. For each of them the history will be updated to track the split
+	 */
+	public static List<ReproducibleInstances> getStratifiedSplit(final ReproducibleInstances data, final Random rand, final double... portions) {
+		return getStratifiedSplit(data, rand.nextLong(), portions);
+	}
+	
+	/**
+	 * Creates a StratifiedSplit for a given {@link ReproducibleInstances} Object. THe History will be updated to track the split.
+	 * 
+	 * @param data - Input data
+	 * @param seed - random seed
+	 * @param portions - ratios to split
+	 * @return a List of {@link ReproducibleInstances}. For each of them the history will be updated to track the split
+	 */
+	public static List<ReproducibleInstances> getStratifiedSplit(final ReproducibleInstances data, final long seed, final double... portions) {
+		Random rand = new Random(seed);
+		/* check that portions sum up to s.th. smaller than 1 */
+		double sum = 0;
+		for (double p : portions) {
+			sum += p;
+		}
+		if (sum > 1) {
+			throw new IllegalArgumentException("Portions must sum up to at most 1.");

```

Comment:
One could also think about normalizing the portions accordingly such that they sum up to 1 in any case. If this is not done, I think it would be a good idea to also check if they sum is smaller than 1.